app/services/dttp/access_token_service.rb

```diff
@@ -0,0 +1,44 @@
+module Dttp
+  class AccessTokenService
+    class Client
+      include HTTParty
+      base_uri "https://login.microsoftonline.com"
+    end
+
+    def self.call
+      new.call
+    end
+
+    def call
+      options = {
+        body: {
+          grant_type: "client_credentials",
+          client_id: client_id,
+          scope: scope,
+          client_secret: client_secret,
+        },
+      }
+
+      response = Client.post("/#{tenant_id}/oauth2/v2.0/token", options)
```

Comment:
Just wondering why we rolled our own and not used something like

https://github.com/oauth-xx/oauth2/wiki/Basic-usage#get-resources

Reply:
It's a `client_credentials` type of OAuth so there didn't seem any point adding the complexity. It's a post request that returns the token. There isn't any of the auth url, redirecting, JWT stuff to worry about.